### ISSUE TITLE
Support for class constructors when used with functools.partial

### ISSUE THREAD

**Issue by LunarLanding** [microsoft/pyright#5559]
**Describe the bug**
A dataclass `__init__` method is not handled correctly by partial.

**To Reproduce**

```py
from dataclasses import dataclass
from functools import partial

@dataclass
class A:
    x:int
    y:int
partial(A,x=1,y='a')
# typechecks and should not    
```

**Expected behavior**
`y should be int` or somesuch.

I suppose more correctly `__call__` should map to `__init__`, and `__init__` is already typed correctly.
With ParamSpec now available in python3.10, this should be easier than before.
https://github.com/microsoft/pyright/issues/1009 is somewhat related.

**Code or Screenshots**
See above.

**VS Code extension or command-line**
VSCode.

**Comment by erictraut** [comment]
This has nothing to do with `dataclass`. Pyright's support for `functools.partial` does not currently work on classes. It works only on functions. In all other cases, it falls back to the normal processing of functools.partial (as it's defined in typeshed), and this does not include validation of argument types.

Adding support for class constructors is a reasonable enhancement request.

**Comment by erictraut** [comment]
This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.319, which I just published. It will also be included in a future release of pylance.
